biocrnpyler/components_membrane.py

```diff
@@ -0,0 +1,198 @@
+        """Initialize a MembraneChannel object to store membrane channel related information.
+        :param product: name of the membrane channel, reference to an Species or Component
+        :param direction: transport direction, taken as "Passive"-undirectional unless specified 
+        :param size: number of monomers needed for channel
+        :param attributes: Species attribute
+        :param keywords: pass into the parent's (Component) initializer
+        """
+        
+        # PROTEIN
+        self.membrane_protein = self.set_species(membrane_protein, material_type='protein', attributes=attributes)
+    
+        # PRODUCT is an integrated membrane protein (transmembrane_protein)
+        if product is None:
+            if direction is None:
+                self.product = self.set_species(product, material_type= 'Passive')
+            else:
+                self.product = self.set_species(product, material_type= direction)
+
+        else:
+            if direction is None:
+                self.product = self.set_species(product, material_type= 'Passive')
+            else:
+                self.product = self.set_species(product, material_type= direction)
+        
+        if size is None:
+            self.membrane_protein.size = 1
+        else:
+            self.membrane_protein.size = size
+        
+        Component.__init__(self=self, name=self.membrane_protein.name, **keywords)
+        
+    def get_species(self):
+        return self.membrane_protein
+    
+    def update_species(self):
+        mech_cat = self.get_mechanism('catalysis')
+        
+        return mech_cat.update_species(self.membrane_protein, self.product)
+
+    def update_reactions(self):
+        mech_cat = self.get_mechanism('catalysis')
+        print(self)
+        
+        return mech_cat.update_reactions(self.membrane_protein, self.product, component=self,  part_id=self.name)
+
+class MembraneChannel(Component):
+    """A class to represent Membrane Channels.
+    Assumes the membrane channel transport substrates in a specific direction across the membrane
+    Uses a mechanism called "catalysis"
+    """
+    def __init__(self, integral_membrane_protein: Union[Species, str, Component],
+                 substrate: Union[Species, str, Component],
+                 direction=None, attributes=None, **keywords):
+        """Initialize a Transporter object to store Transport membrane related information.
+        :param substrate: name of the substrate, reference to an Species or Component
+        :param direction: give direction of transport ref to vesicle
+        :param attributes: Species attribute
+        :param keywords: pass into the parent's (Component) initializer
+        """
+
+   # SUBSTRATE
+        if substrate is None:
+            self.substrate = None
+        else:
+            product=substrate
+            self.substrate = self.set_species(substrate, compartment='Internal',attributes=attributes)
```

Comment:
In the case of membrane components, the substrate is the substance on which the transporter/channel acts without distinction of compartment. The molecule does not change because the substrate and product are the same substance; only the compartment of the substance changes. Thus, the product here is based on the action of the transporter. I can add this as a comment to the code for clarity.